lib/plugin.js

```diff
@@ -94,20 +94,16 @@ ManifestPlugin.prototype.apply = function(compiler) {
 
     var json = JSON.stringify(cache, null, 2);
 
-    compilation.assets[outputName] = {
-      source: function() {
-        return json;
-      },
-      size: function() {
-        return json.length;
-      }
-    };
-
-    if (this.opts.writeToFileEmit) {
-      var outputFolder = compilation.options.output.path;
-      var outputFile = path.join(outputFolder, this.opts.fileName);
+    var outputFolder = compilation.options.output.path;
+    var outputFile = path.join(outputFolder, this.opts.fileName);
 
+    const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem' ||
+                                      this.opts.writeToFileEmit;
+    if (shouldUseNodeFs) {
       fse.outputFileSync(outputFile, json);
+    } else {
+      compiler.outputFileSystem.mkdirpSync(path.dirname(outputFile));
```

Comment:
I wonder why `mkdirpSync` and not `mkdirp`?
Webpack docs only mention `mkdirp` [here](https://webpack.js.org/api/node/#custom-file-systems) so it might blow up for some custom FS that provides only `mkdirp` but doesn't provide `mkdirpSync`